src/Settings/WalletSettings/WalletSettingsScreen.tsx

```diff
@@ -100,10 +111,86 @@ export const WalletSettingsScreen = () => {
           value={intl.formatMessage(getWalletType(wallet.walletImplementationId))}
         />
       </SettingsSection>
+      <LoadingOverlay loading={isResyncLoading} />
     </ScrollView>
   )
 }
 
+const getNetworkName = (networkId: NetworkId) => {
+  // note(v-almonacid): this throws when switching wallet
+  try {
+    const config = getNetworkConfigById(networkId)
+    return config.MARKETING_NAME
+  } catch (_e) {
+    return '-'
+  }
+}
+
+const getWalletType = (implementationId: WalletImplementationId): MessageDescriptor => {
+  if (isByron(implementationId)) return messages.byronWallet
+  if (isHaskellShelley(implementationId)) return messages.shelleyWallet
+
+  return messages.unknownWalletType
+}
+
+const LogoutButton = () => {
+  const strings = useStrings()
+  const {logoutWithConfirmation, isLoading} = useLogout()
+
+  return <PressableSettingsItem label={strings.logout} onPress={logoutWithConfirmation} disabled={isLoading} />
+}
+
+const ResyncButton = ({onClick, isLoading}: {onClick: () => void; isLoading: boolean}) => {
+  const strings = useStrings()
+
+  return <PressableSettingsItem label={strings.resync} onPress={onClick} disabled={isLoading} />
+}
+
+const LoadingOverlay = ({loading}: {loading: boolean}) => {
```

Comment:
why was this added?

Reply:
Now the resync action is called when the user is still in settings and when the resync is done the user goes back to tx list.

Reply:
That is the reason the loading variable is used in that way.

Reply:
why not navigate immediately?

Reply:
why doesnt it load smoothly?